src/healthkit.cs

```diff
@@ -3105,4 +3371,84 @@ interface HKHeartbeatSeriesQuery
 		[DesignatedInitializer]
 		IntPtr Constructor (HKHeartbeatSeriesSample heartbeatSeries, HKHeartbeatSeriesQueryDataHandler dataHandler);
 	}
+
+	[Watch (7,0), iOS (14,0)]
+	[BaseType (typeof (HKSample))]
+	[DisableDefaultCtor] // NSInvalidArgumentException Reason: The -init method is not available on HKElectrocardiogram
+	interface HKElectrocardiogram
+	{
+		[Export ("numberOfVoltageMeasurements")]
+		nint NumberOfVoltageMeasurements { get; }
+
+		[NullAllowed, Export ("samplingFrequency", ArgumentSemantic.Copy)]
+		HKQuantity SamplingFrequency { get; }
+
+		[Export ("classification", ArgumentSemantic.Assign)]
+		HKElectrocardiogramClassification Classification { get; }
+
+		[NullAllowed, Export ("averageHeartRate", ArgumentSemantic.Copy)]
+		HKQuantity AverageHeartRate { get; }
+
+		[Export ("symptomsStatus", ArgumentSemantic.Assign)]
+		HKElectrocardiogramSymptomsStatus SymptomsStatus { get; }
+	}
+
+	delegate void HKElectrocardiogramQueryDataHandler (HKElectrocardiogramQuery query, HKElectrocardiogramVoltageMeasurement voltageMeasurement, bool done, NSError error);
+
+	[Watch (7,0), iOS (14,0)]
+	[BaseType (typeof (HKQuery))]
+	interface HKElectrocardiogramQuery
+	{
+
+		[Export ("initWithElectrocardiogram:dataHandler:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (HKElectrocardiogram electrocardiogram, HKElectrocardiogramQueryDataHandler dataHandler);
+	}
+
+	[Watch (7,0), iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	interface HKElectrocardiogramVoltageMeasurement
+	{
+		[Export ("timeSinceSampleStart")]
+		double TimeSinceSampleStart { get; }
+
+		[Export ("quantityForLead:")]
+		[return: NullAllowed]
+		HKQuantity QuantityForLead (HKElectrocardiogramLead lead);
+	}
+
+	[NoWatch, iOS (14,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface HKFHIRVersion : NSCopying, NSSecureCoding
+	{
+		[Export ("majorVersion")]
+		nint MajorVersion { get; }
+
+		[Export ("minorVersion")]
+		nint MinorVersion { get; }
+
+		[Export ("patchVersion")]
+		nint PatchVersion { get; }
+
+		[Export ("FHIRRelease", ArgumentSemantic.Strong)]
+		string FHIRRelease { get; }
+
+		[Export ("stringRepresentation")]
+		string StringRepresentation { get; }
+
+		[Static]
+		[Export ("versionFromVersionString:error:")]
+		[return: NullAllowed]
+		HKFHIRVersion VersionFromVersionString (string versionString, [NullAllowed] out NSError errorOut);
+
+		[Static]
+		[Export ("primaryDSTU2Version")]
+		HKFHIRVersion PrimaryDSTU2Version ();
```

Comment:
Add `Get*`
or turn it into a property (look what swift is doing)

Reply:
It makes sense to change these into properties since they're more or less getters - no way to modify, no params, returns a single value.

Can you explain more about what Swift is doing?

I took a look at the docs: https://docs.developer.st.apple.com/documentation/healthkit/hkfhirversion

As well as the objc header code.

Reply:
> It makes sense...

Yes, those are all good points to make a property over a method 😄 

There are also _less visible_ aspects that are important. One of them is performance, IOW how costly is it to call the code. People expect properties to be _cheap_ but it's hard to know since we don't have the source code for the implementation. E.g.

```csharp
long total = 0;
// with a method most people will realize there's a cost and call it once, out of the loop
byte[] large = foo.GetBlob ();
foreach (byte b in large)
    total += b;
```

```csharp
long total = 0;
// for properties it's more fuzzy since it's unclear if the result is cached or not...
for (int i =0 < i; foo.Blob.Length; i++)
    total += foo.Blob [i];
```

if you end up allocating a new `byte[]` for each iteration then you'll have a performance bottleneck on the property version.

Looking at what Swift does gives us a _hint_ if Apple thinks it's a good candidate to be a property (even if it's not Ibn ObjC) versus a method.

> Can you explain more about what Swift is doing?

`class func primaryDSTU2() -> Self`

TL&DR: It's not a property in Swift either.

Longer story: ObjC did not always had properties and even some _recent_ code/headers did not use them. Looking at Swift gives you a second, often better, opinion of the API.